devcade-library/Input.cs

```diff
@@ -120,16 +127,35 @@ public static void Initialize()
       p1LastState = GamePad.GetState(0);
       p2LastState = GamePad.GetState(1);
     }
+    
+    internal static void UpdateInternal() {
+      internalUpdate = true;
+      if (externalUpdate) {
+        throw new Exception("Cannot use Input.Update() and InputManager.Update() in the same project");
+      }
+      p1LastState = p1State;
+      p2LastState = p2State;
+      p1State = GamePad.GetState(0);
+      p2State = GamePad.GetState(1);
+    }
 
     /// <summary>
     /// Updates input states.
     /// </summary>
-    public static void Update()
-    {
+    public static void Update() {
+      externalUpdate = true;
+      if (internalUpdate) {
+        throw new Exception("Cannot use Input.Update() and InputManager.Update() in the same project");
+      }
       p1LastState = p1State;
       p2LastState = p2State;
       p1State = GamePad.GetState(0);
       p2State = GamePad.GetState(1);
     }
+    
+    public static (GamePadState, GamePadState) GetStates()
```

Comment:
I just learned that you could do this returning multiple variables, but should we not name the variables in the tuple?

Reply:
variables contained in tuples cannot be named, it's essentially an array of length 2 so the members are accessed by .0 and .1, iirc.

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples#tuple-field-names